PowerEditor/src/MISC/Common/FileInterface.cpp

```diff
@@ -19,28 +19,32 @@
 #include "FileInterface.h"
 #include "Parameters.h"
 
-Win32_IO_File::Win32_IO_File(const char *fname)
-{
-	if (fname)
-	{
-		_path = fname;
-		_hFile = ::CreateFileA(fname, _accessParam, _shareParam, NULL, _dispParam, _attribParam, NULL);
-	}
-}
-
 
 Win32_IO_File::Win32_IO_File(const wchar_t *fname)
 {
 	if (fname)
 	{
-		std::wstring fn = fname;
-		std::wstring_convert<std::codecvt_utf8_utf16<wchar_t>> converter;
-		_path = converter.to_bytes(fn);
-		_hFile = ::CreateFileW(fname, _accessParam, _shareParam, NULL, _dispParam, _attribParam, NULL);
+		static constexpr DWORD accessParam	= GENERIC_READ | GENERIC_WRITE;
+		static constexpr DWORD shareParam	= FILE_SHARE_READ | FILE_SHARE_WRITE;
+		static constexpr DWORD attribParam	= FILE_ATTRIBUTE_NORMAL;
+
+		_hFile = ::CreateFileW(fname, accessParam, shareParam, NULL, TRUNCATE_EXISTING, attribParam, NULL);
+
+		if (_hFile == INVALID_HANDLE_VALUE && ::GetLastError() == ERROR_FILE_NOT_FOUND)
+		{
+			_hFile = ::CreateFileW(fname, accessParam, shareParam, NULL, CREATE_NEW, attribParam, NULL);
+
+			if (_hFile == INVALID_HANDLE_VALUE && ::GetLastError() == ERROR_FILE_EXISTS)
+				_hFile = ::CreateFileW(fname, accessParam, shareParam, NULL, TRUNCATE_EXISTING, attribParam, NULL);
+		}

```

Comment:
We discussed this. It startet here: https://github.com/notepad-plus-plus/notepad-plus-plus/pull/14039#discussion_r1302573153
@pnedev did what he commented to that first message. Here: https://github.com/notepad-plus-plus/notepad-plus-plus/pull/14039#discussion_r1302612004

Reply:
@Klotzi111 
> You could change your code to first try to open the file with TRUNCATE_EXISTING and if that fails with last error = ERROR_FILE_NOT_FOUND you then go on and try open with CREATE_NEW.

I did understand this part.

> If now that fails with last error = ERROR_FILE_EXISTS you could try TRUNCATE_EXISTING again.

But in which circumstance we will go into this case?

Reply:
@donho ,
The second TRUNCATE_EXISTING is called as a insurance that between CREATE_NEW and TRUNCATE_EXISTING another program hasn't created the file.

BR

Reply:
@pnedev 
First calling with TRUNCATE_EXISTING fail when the file doesn't exist.
So we call again with CREATE_NEW - at this moment it will and always be successful since in current master only CREATE_NEW is used, so I don't see how it will fail because file exist.

Reply:
@donho ,
Here is a very unlikely scenario but still possible:
1) TRUNCATE_EXISTING fails -> File doesn't exist.
2) External program A creates the file.
3) CREATE_NEW fails -> File exists. (CREATE_NEW is not CREATE_ALWAYS that we used so far in the code)
4) Second TRUNCATE_EXISTING is needed -> it will succeed in that case

Once again - you can modify the PR as you like.
BR

Reply:
@pnedev 
> (CREATE_NEW is not CREATE_ALWAYS that we used so far in the code).

My bad. You're right.
Thank you for the explanation.

Reply:
@pnedev 
OK, I see your point.
However, it's kind of excessive to me for calling 3 times CreateFileW.
I keep my PR anyway to see the test result (if any), then we'll see.